components/eamxx/src/share/field/field_alloc_prop.cpp

```diff
@@ -161,31 +161,33 @@ void FieldAllocProp::commit (const layout_ptr_type& layout)
   // Store pointer to layout for future use (in case subview is called)
   m_layout = layout;
 
-  // Loop on all value type sizes.
-  m_last_extent = 0;
-  int last_phys_extent = m_layout->dims().back();
-  for (auto vts : m_value_type_sizes) {
-    // The number of scalar_type in a value_type
-    const int vt_len = vts / m_scalar_type_size;
+  if (m_layout->size()==0) {
+    // Zero-dimensional fields are supported, so we may end up with a
+    // layout of size 0. In this case allocate a single scalar, but
+    // set the last extent to 0 since we have no dimension.
+    m_alloc_size = m_scalar_type_size;
```

Comment:
What if the user (for some reason) asked for this field to accommodate packing? In that case `m_alloc_size = m_scalar_type_size` seems wrong...

OTOH, I don't think anyone will ever ask for a rank-0 field that is packed, so we could just error out in that case? So perhaps just add a check that `m_value_type_sizes` has size 1 (we always have `m_scalar_type_size` in it)?

Reply:
Actually, we _might_ hit that case. We can create a rank-0 subfield of a packed field. In that case, we may have to allow a non-trivial pack size...

Reply:
Yeah, padding is probably an issue. And we probably don't just want to error out in case we are calling `get_padding()` on multiple fields, not just ones we think will have padding. Could be annoying.

Something I was thinking about as well is subviewing an already subviewed 1d field (or any case where we aren't guaranteed contiguous data). Maybe that could be added to the tests if we think it works (or maybe I'm overthinking).